include/boost/thread/executors/serial_executor_cont.hpp

```diff
@@ -32,7 +39,7 @@ namespace executors
   private:
 
     generic_executor_ref ex_;
-    future<void> fut_; // protected by mtx_
+    BOOST_THREAD_FUTURE<void> fut_; // protected by mtx_
```

Comment:
This  is not absolutely necessary as this header is supported only after version 3.
Anyway, it doesn't hurt.

Reply:
I did not see this documented anywhere, nor guarded by any macro checks. There must be a macro check so that the header can be included by the tests.

Reply:
Unfortunately BOOST_THREAD_PROVIDES_EXECUTORS is not defined in any BOOST_THREAD_VERSION
```
Feature                             | Anti-Feature                     | V2   | V3  | V4
PROVIDES_EXECUTORS | -                                         | NO  | NO | NO
PROVIDES_FUTURE        | DONT_PROVIDE_FUTURE | NO | YES | YES
```

https://www.boost.org/doc/libs/master/doc/html/thread/build.html#thread.build.configuration

Reply:
Yes, I noticed that.